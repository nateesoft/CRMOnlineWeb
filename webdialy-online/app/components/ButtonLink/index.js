/**
 *
 * ButtonLink
 *
 */

import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledLink = styled(Link)`
  text-decoration: none;
  color: ${props => (props.color ? props.color : 'black')};
`;

function ButtonLink(props) {
  return <StyledLink {...props}>{props.children}</StyledLink>;
}

ButtonLink.propTypes = {
  children: PropTypes.any,
};

export default ButtonLink;
