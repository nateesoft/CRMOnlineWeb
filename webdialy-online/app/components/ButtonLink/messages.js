/*
 * ButtonLink Messages
 *
 * This contains all the text for the ButtonLink component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.ButtonLink';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the ButtonLink component!',
  },
});
