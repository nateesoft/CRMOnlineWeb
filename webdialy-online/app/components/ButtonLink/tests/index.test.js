import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

import ButtonLink from '../index';

describe('ButtonLink components', () => {
  it('test snapshot', () => {
    const component = renderer.create(
      <MemoryRouter>
        <ButtonLink to="/">Test Button</ButtonLink>
      </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
