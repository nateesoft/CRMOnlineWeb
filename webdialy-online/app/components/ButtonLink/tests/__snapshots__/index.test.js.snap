// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink components test snapshot 1`] = `
<a
  className="ButtonLink__StyledLink-sc-1pptk3-0 gOeyQX"
  href="/"
  onClick={[Function]}
>
  Test Button
</a>
`;
